modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/util/ObjectEntryVariablesUtil.java

```diff
@@ -211,7 +211,7 @@ private static Map<String, Object> _getVariables(
 						(Map<String, Object>)payloadJSONObject.get(
 							"objectEntry");
 
-					return objectEntry.get("userId");
```

Comment:
@DaneszLR please check if we can change this code to something like:

```
long userId = payloadJSONObject.getLong("userId");

Map<String, Object> allowedVariables =
	HashMapBuilder.<String, Object>put(
		"creator",
		() -> {
			if (objectDefinition.isUnmodifiableSystemObject()) {
				return userId;
			}

			return MapUtil.getLong(
				(Map<String, Object>)payloadJSONObject.get(
					"objectEntry"),
				"userId");
		}
	).put(
		"currentUserId", userId
	).build();

```

This way we can always guarantee the same type for the `userId` property, and it makes more sense to me if it's `Long`.

Reply:
@DaneszLR the idea is to make sure that we always return the same type for both cases, avoiding bugs like this one, since `objectEntry.get` and `payloadJSONObject.get` return a `Object`. We can also keep the `String` type:

```
		String userId = payloadJSONObject.getString("userId");

		Map<String, Object> allowedVariables =
			HashMapBuilder.<String, Object>put(
				"creator",
				() -> {
					if (objectDefinition.isUnmodifiableSystemObject()) {
						return userId;
					}

					return MapUtil.getString(
						(Map<String, Object>)payloadJSONObject.get(
							"objectEntry"),
						"userId");
				}
			).put(
				"currentUserId", userId
			).build();
```